src/formatDate.js

```diff
@@ -49,8 +49,34 @@
  * @returns {string}
  */
 
+const formatYear = (date) => {
+  const CURRENT_YEAR = Number(`${new Date().getFullYear()}`.slice(-2));
+  const newDate = { ...date };
+
+  if (Object.prototype.hasOwnProperty.call(newDate, 'YYYY')) {
+    newDate['YY'] = newDate['YYYY'].slice(-2);
+  } else {
+    newDate['YYYY'] = newDate['YY']
+      .padStart(4, newDate['YY'] < CURRENT_YEAR ? '20' : '19');
+  }
+
+  return newDate;
+};
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitDate = date.split(fromFormat.slice(-1));
+  const dates = splitDate.reduce((obj, unit, i) => {
+    obj[fromFormat[i]] = unit;
+
+    return obj;
+  }, {});
+
+  // Лучше мутировать этот объект или делать копию как сейчас?
+  const yearFormated = formatYear(dates);
+
```

Comment:
`dates` is an internal object of the function - you can do whatever you want

If I were you - I probably did not bother to put any additional effort to mutate or copy it. So if your solution looks natural mutating - let it be so.